scheduler-pool/benches/scheduler.rs

```diff
@@ -0,0 +1,894 @@
+                tx_account_locks
+                    .push(tx.get_account_locks(bank.get_transaction_account_lock_limit())?);
+            }
+
+            // the dependency graph contains the indices that must be executed (marked with
+            // State::Done) before they can be executed
+            let dependency_graph = Self::build_dependency_graph(&tx_account_locks);
+
+            #[derive(Clone)]
+            enum State {
+                Blocked,
+                Processing,
+                Done,
+            }
+
+            let mut processing_states: Vec<State> = vec![State::Blocked; dependency_graph.len()];
+            let mut signature_indices: HashMap<&Signature, usize> =
+                HashMap::with_capacity(dependency_graph.len());
+            signature_indices.extend(
+                pending_transactions
+                    .iter()
+                    .enumerate()
+                    .map(|(idx, tx)| (tx.signature(), idx)),
+            );
+
+            let mut is_done = false;
+            while !is_done {
+                is_done = true;
+                for idx in 0..processing_states.len() {
+                    match processing_states[idx] {
+                        State::Blocked => {
+                            is_done = false;
+
+                            // if all the dependent txs are executed, this transaction can be
+                            // scheduled for execution.
+                            if dependency_graph[idx]
+                                .iter()
+                                .all(|idx| matches!(processing_states[*idx], State::Done))
+                            {
+                                self.inner_scheduelr.schedule_execution(Arc::new((
+                                    pending_transactions[idx].clone(),
+                                    idx,
+                                )));
+                                // this idx can be scheduled and moved to processing
+                                processing_states[idx] = State::Processing;
+                            }
+                        }
+                        State::Processing => {
+                            is_done = false;
+                        }
+                        State::Done => {}
+                    }
+                }
+
+                if is_done {
+                    break;
+                }
+
+                let mut executor_responses: Vec<_> = vec![receiver.recv().unwrap()];
+                executor_responses.extend(receiver.try_iter());
+                for r in &executor_responses {
+                    processing_states[*signature_indices.get(r).unwrap()] = State::Done;
+                }
+            }
+            Ok(())
+        }
+    }
+
+    // frequent synchronizations hampers efficient (= parallelizable) scheduling of several chunks
+    // of txes which are tied together for each common account locks. Ideally those independent chunks can be
```

Comment:
Consider starting comment from a capital letter.
Otherwise, it looks like it is missing a line or more.
You are using capital letters to start all subsequent sentences.